### PR TITLE
Add targetElement prop

### PR DESCRIPTION
**Pull request by edellucien**
# How to

 - Add copyright example
 - Optional parameter 'targetElement' added to the copyright control.
 - Add 'mobilityControls' option to add controls to map
# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [ ] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [x] The new class' members & methods are well documented
- [ ] Tests added.

### CODE REVIEW

File: src/common/controls/Control.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Control extends Observable {
   constructor(map, options = {}) {
     super();
     this.map = map;
+    this.targetElement = options.targetElement || this.map.getTargetElement();
```
Comment (edellucien) [inline src/common/controls/Control.js:17]: Do you agree on setting the map target element as default target for the copyright?

File: src/ol/Map.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import OLLayer from 'ol/layer/Layer';
 import { defaults as defaultControls } from 'ol/control';
 import Layer from './layers/Layer';
 import mixin from '../common/mixins/MapMixin';
-import CopyrightControl from './controls/Copyright';
+// import CopyrightControl from './controls/Copyright';
```
Comment (edellucien) [inline src/ol/Map.js:6]: Do we actually still want to set CopyrightControl as default like this?

File: src/doc/examples/ol-copyright.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  map.addMobilityControl(customCopyrightControl);
+
+  const url = `https://maps.geops.io/styles/base_bright_v2/style.json?key=${window.apiKey}`;
+  const mapboxLayer = new MapboxLayer({ url });
+
+  map.addLayer(mapboxLayer);
+
+  /*
+  // later, you can remove the control
```
Comment (edellucien) [inline src/doc/examples/ol-copyright.js:29]: We could think of a button to remove the copyright. In order to show how to do it.

File: src/common/controls/Control.js [hunk 4]
```diff
@@ -10,10 +10,12 @@ class Control extends Observable {
    * @param {ol/Map~Map|mapboxgl.Map} map Control's map.
    * @param {Object} [options] Control options.
    * @param {boolean} [options.active = true] Whether the control is active.
+   * @param {HTMLElement} [options.targetElement = map.getTargetElement()] Container element where to locate the copyright.
    */
   constructor(map, options = {}) {
     super();
     this.map = map;
+    this.targetElement = options.targetElement || this.map.getTargetElement();
```
Comment (mhaertwig) [inline src/common/controls/Control.js:18]: `getTargetElement()` only works for OpenLayers maps, not for a mapbox map. So, please remove everything behind the `||` and put `this.target = options.targetElement` in the constructor of the mixin.

File: src/common/controls/mixins/Copyright.js [hunk 5]
```diff
@@ -1,9 +1,8 @@
 const CopyrightMixin = (Base) =>
   class extends Base {
-    addCopyrightContainer(target) {
-      this.target = target;
```
Comment (mhaertwig) [inline src/common/controls/mixins/Copyright.js:4]: Do `this.target = this.target || target`;

**Review by mhaertwig — CHANGES_REQUESTED** [review]

**Comment by edellucien** [comment]
Code adapted as you wished

File: src/common/controls/Control.js [hunk 6]
```diff
@@ -10,10 +10,12 @@ class Control extends Observable {
    * @param {ol/Map~Map|mapboxgl.Map} map Control's map.
    * @param {Object} [options] Control options.
    * @param {boolean} [options.active = true] Whether the control is active.
+   * @param {HTMLElement} [options.targetElement = map.getTargetElement()] Container element where to locate the copyright.
    */
   constructor(map, options = {}) {
     super();
     this.map = map;
+    this.target = options.targetElement;
```
Comment (mhaertwig) [inline src/common/controls/Control.js:18]: Please remove both lines in this file (see next comment).

File: src/common/controls/mixins/Copyright.js [hunk 7]
```diff
@@ -1,9 +1,9 @@
 const CopyrightMixin = (Base) =>
   class extends Base {
     addCopyrightContainer(target) {
-      this.target = target;
+      this.target = this.target || target;
```
Comment (mhaertwig) [inline src/common/controls/mixins/Copyright.js:4]: Do `this.target = this.options.targetElement || target`.

File: src/ol/controls/Copyright.js [hunk 8]
```diff
@@ -17,7 +17,7 @@ class Copyright extends mixin(CommonControl) {
 
   activate() {
     super.activate();
-    this.addCopyrightContainer(this.map.getTargetElement());
+    this.addCopyrightContainer();
```
Comment (mhaertwig) [inline src/ol/controls/Copyright.js:20]: The original one was correct. You have to pass the container as a function.

**Review by mhaertwig — CHANGES_REQUESTED** [review]

File: src/common/controls/mixins/Copyright.js [hunk 9]
```diff
@@ -1,9 +1,19 @@
+/* eslint-disable no-useless-constructor */
```
Comment (edellucien) [inline src/common/controls/mixins/Copyright.js:1]:
```suggestion
```

**Review by oterral — APPROVED** [review]
lgtm

**Pull request merged**